src/common/hist_util.cc

```diff
@@ -749,7 +866,12 @@ struct Prefetch {
 
 constexpr size_t Prefetch::kNoPrefetchSize;
 
-template<typename FPType, bool do_prefetch>
+template <typename T>
+static constexpr size_t GetPrefetchStep() {
```

Comment:
I think you need to move this to the class again. I don't see reasons to make this non-member function.

Reply:
moved